File: patterns2-infoauth/patterns2-infoauth/Model/UserCreationDto.cs
using System.ComponentModel.DataAnnotations;

namespace patterns2_infoauth.Model
{
    public class UserCreationDto
    {
        [Phone]
        public string Phone { get; set; }
        [Length(8, 32)]
        public string Password { get; set; }
        public string Name { get; set; }
        public RoleType? Role { get; set; }
    }
}

File: client-app/Hbank/app/src/main/java/com/example/h_bank/data/dto/UserCreationDto.kt
package com.example.h_bank.data.dto

import com.example.h_bank.data.RoleType
import kotlinx.serialization.Serializable

@Serializable
class UserCreationDto(
    val phone: String,
    val password: String,
    val name: String,
    val role: RoleType?
)